p2panda-stream/src/stream/decode.rs

```diff
@@ -103,15 +103,27 @@ where
 mod tests {
     use futures_util::{StreamExt, TryStreamExt};
     use p2panda_core::{Body, Header};
+    use serde::{Deserialize, Serialize};
 
     use crate::test_utils::mock_stream;
 
     use super::DecodeExt;
 
     #[tokio::test]
     async fn decode() {
+        #[derive(Clone, Debug, Serialize, Deserialize)]
+        struct MyExtensions {}
+
+        let stream = mock_stream().decode();
+        let result: Vec<(Header<MyExtensions>, Option<Body>, Vec<u8>)> =
+            stream.take(5).try_collect().await.expect("not fail");
+        assert_eq!(result.len(), 5);
+    }
+
+    #[tokio::test]
+    async fn decode_non_map_extensions() {
         let stream = mock_stream().decode();
-        let result: Vec<(Header, Option<Body>, Vec<u8>)> =
+        let result: Vec<(Header<()>, Option<Body>, Vec<u8>)> =
             stream.take(5).try_collect().await.expect("not fail");
         assert_eq!(result.len(), 5);
     }
```

Comment:
@adzialocha this is the failing test, it doesn't like that the extensions are a `()`, it fails with the error: `Semantic(None, "invalid extensions: Semantic(None, \"invalid type: map, expected unit\")")`

Reply:
Thing that confuses me is that I believe I added a test for decoding an operation containing similar extensions here: https://github.com/p2panda/p2panda/blob/6650b351f2141b542f3e8aa9f86eb958920ce2a0/p2panda-core/src/serde.rs#L746-L765 and it passes :thinking:

Reply:
The reason this got exposed at all is because I removed the `DefaultExtensions` struct and updated the tests.

Reply:
`mock_stream` already comes with it's own extension format: https://github.com/p2panda/p2panda/blob/v2/p2panda-stream/src/test_utils.rs#L53-L56

You can make it generic to make your test work!

Reply:
I removed the test as there is a test for exactly the same decoding case in `p2panda_core::serde`